### ISSUE TITLE
TYPE and SCAN commands are inconsistent with Redis

### ISSUE THREAD

**Issue by caipengbo** [apache/kvrocks#928]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

Release 2.1.0

### Minimal reproduce step

1. [Type command](https://redis.io/commands/type/) return simple string in redis, but return bulk string in kvrocks.
2. [Scan command](https://redis.io/commands/scan/) can return empty key in redis, but return none in kvrocks.

### What did you expect to see?

These two commands should be consistent with Redis.

```
127.0.0.1:6379> set "" value
OK

127.0.0.1:6379 > TYPE ""
string

127.0.0.1:6379 > scan 0
1) "0"
2) 1) ""
```

### What did you see instead?

The semantics of the Type and Scan commands are inconsistent with redis.

```
127.0.0.1:8888> set "" value
OK

127.0.0.1:8888 > TYPE key
"string"

127.0.0.1:8888 > scan 0
1) "0"
2) 1) (nil)
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
Agree. Also, SCAN returns "_cursor" instead of "cursor" on Redis, which causes #842 cannot directly make use of `redis.Client#Scan`.

**Comment by enjoy-binbin** [comment]
this was fixed in #1269

**Comment by git-hulk** [comment]
@enjoy-binbin Thanks for your warm reminder, closing...
